### PR TITLE
Add additional scheme diagnostic options

### PR DESCRIPTION
**Pull request by navartis**
Resolves https://github.com/tuist/tuist/issues/5212

### Short description 📝

Add support of additional scheme diagnostic options.

### How to test the changes locally 🧐

Use  .enableAddressSanitizer, .enableDetectStackUseAfterReturn, .enableThreadSanitizer 
cases of SchemeDiagnosticsOption for enabling Address Sanitizer, Detect use of stack after return, Thread Sanitizer options.

> Include a set of steps for the reviewer to test the changes locally (see [the documentation](https://docs.tuist.io/contributors/testing-strategy) for reference).

### Contributor checklist ✅

- [ ] The code has been linted using run `./fourier lint tuist --fix`
- [ ] The change is tested via unit testing or acceptance testing, or both
- [ ] The title of the PR is formulated in a way that is usable as a changelog entry
- [ ] In case the PR introduces changes that affect users, the documentation has been updated

### Reviewer checklist ✅

- [ ] The code architecture and patterns are consistent with the rest of the codebase
- [ ] Reviewer has checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`, and the title is usable as a changelog entry

### CODE REVIEW

**Comment by pepicrft** [comment]
@navartis you'll have to run `./fourier lint tuist --fix` to fix the linting issues and push the changes upstream. Make sure you install dependencies with `bundle install` beforehand. Please, let me know if it doesn't work.

File: Sources/ProjectDescription/SchemeDiagnosticsOption.swift [hunk 1]
```diff
@@ -4,6 +4,15 @@ import Foundation
 
 /// Options to configure scheme diagnostics for run and test actions.
 public enum SchemeDiagnosticsOption: String, Equatable, Codable {
+    /// Enable the address sanitizer
+    case enableAddressSanitizer
+    
+    /// Enable the detect use of stack after return of address sanitizer
+    case enableASanStackUseAfterReturn
```
Comment (kwridan) [inline Sources/ProjectDescription/SchemeDiagnosticsOption.swift:11]: Perhaps we can leverage a better name here - this doesn't need to map 1-1 with the underlying xcscheme names

 ```suggestion
    case enableDetectStackUseAfterReturn
```

**Review by kwridan — COMMENTED** [review]
Thanks @navartis

**Comment by navartis** [comment]
> @navartis you'll have to run `./fourier lint tuist --fix` to fix the linting issues and push the changes upstream. Make sure you install dependencies with `bundle install` beforehand. Please, let me know if it doesn't work.

I did it!

But after linting command execution I have changes in `projects/tuist/fixtures/tuist_plugin` git submodule.
It seems to be the right thing to ignore them. Correct me if I'm wrong.

**Comment by pepicrft** [comment]
> I am pleasantly surprised by such quick review!
> I have never met such active contributors in public projects.
> Hope my trivial PR will benefit the entire community;)

The community is always thankful for people like you going ahead and helping us improve the project. Thanks again and keep those great contributions coming :)

**Pull request merged**